packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -698,11 +698,24 @@ describe('<TreeItem />', () => {
           expect(getByTestId('one')).to.have.attribute('aria-selected', 'false');
           fireEvent.keyDown(document.activeElement, { key: ' ' });
           expect(getByTestId('one')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('one')).to.have.class('Mui-selected');
+        });
+
+        it('should not select a node when space is pressed and disableSelection', () => {
+          const { getByTestId } = render(
+            <TreeView disableSelection>
+              <TreeItem nodeId="one" label="one" data-testid="one" />
+            </TreeView>,
+          );
+
+          getByTestId('one').focus();
+          fireEvent.keyDown(document.activeElement, { key: ' ' });
+          expect(getByTestId('one')).not.to.have.class('Mui-selected');
```

Comment:
Prefer checking for `aria-selected` everywhere. Makes it easier to "grep" the codebase. I don't think we need to test for the classname on every check.

Reply:
I deliberately chose the class over aria-selected as the aria-selected logic is incorrect for single select https://github.com/mui-org/material-ui/issues/20099.  

> If the tree does not support multiple selection, aria-selected is set to true for the selected node and it is **not present on any other node in the tree.**

If you accept the [pull request](https://github.com/mui-org/material-ui/pull/20102) then the test would have to change from testing for aria-selected false to absence of aria-selected.

Reply:
New commit uses aria-selected.